docs/dev/cut-a-new-release.md

```diff
@@ -1,64 +1,230 @@
+> For some developer workflows, `upstream` can be the same as `origin`.
+
+**Table of Contents**:
+
+* [Development](#development)
+* [Regular release](#regular-release)
+* [Hotfix release for latest version](#hotfix-release-for-latest-version)
+* [Hotfix release for previous version](#hotfix-release-for-previous-version)
+
+## Published artifacts
+
+* The Kotlin libraries are published: [GitHub Releases](https://github.com/mozilla/glean/releases), [Mozilla Maven](https://maven.mozilla.org/?prefix=maven2/org/mozilla/telemetry/).
+* Python bindings are published on PyPI: [glean-sdk](https://pypi.org/project/glean-sdk/).
+* iOS framework artifacts: [GitHub Releases](https://github.com/mozilla/glean/releases).
+* Rust crates are published on crates.io: [glean-core](https://crates.io/crates/glean-core), [glean-ffi](https://crates.io/crates/glean-ffi).
+
+## Development
+
+All finished features for Glean land on the `master` branch.
+Features should be developed in feature branches and are sent as pull requests to the upstream repository.
+
+1. Clone the Glean repository:
+    ```
+    git clone https://github.com/your-name/glean && cd glean
+    ```
+2. Check out a new feature branch:
+    ```
+    git checkout -b your-feature-branch
+    ```
+3. Implement your feature and add information about it to `CHANGELOG.md` under the `Unreleased changes` headline.
+4. Push your branch:
+    ```
+    git push origin your-feature-branch
+    ```
+5. Create a new pull request: <https://github.com/mozilla/glean/compare/master...your-name:your-feature-branch?expand=1>
+
+The Glean maintainers will review your code and merge it when ready.
+Make sure all tests pass.
+
+## Regular Release
+
+Releases can only be done by one of the Glean maintainers.
+
+* Main development branch: `master`
+* Main release branch: `release`
+* Specific release branch: `release-vX.Y.Z`
+* Hotfix branch: `hotfix-X.Y.(Z+1)`
+
+### Create a release branch
+
+1. Create a release branch from the `master` branch:
+    ```
+    git checkout -b release-v25.0.0 master
+    ```
+2. Update the changelog .
     1. Add any missing important changes under the `Unreleased changes` headline.
     2. Commit any changes to the changelog file due to the previous step.
-2. Run `bin/prepare-release.sh <new version>`
+3. Run `bin/prepare-release.sh <new version>` to bump the version number.
     1. The new version should be the next patch, minor or major version of what is currently released.
     2. Let it create a commit for you.
-3. Land the new commit that performs the steps above. This takes a PR, typically, because of branch protection on master.
-4. Cut the actual release.
-    1. Click "Releases", and then "Draft a New Release" in the GitHub UI.
+4. Push the new release branch:
+    ```
+    git push upstream release-v25.0.0
+    ```
+5. Wait for CI to finish on that branch and ensure it's green: <https://circleci.com/gh/mozilla/glean/tree/release-v25.0.0>
```

Comment:
For bonus points, the `prepare-release.sh` script could print out these two instructions at the end :)  Having a clickable link to circleci would be particularly handy.